dagfactory/dagbuilder.py

```diff
@@ -624,6 +610,104 @@ def replace_expand_values(task_conf: Dict, tasks_dict: Dict[str, BaseOperator]):
                     task_conf["expand"][expand_key] = tasks_dict[task_id].output
         return task_conf
 
+    @staticmethod
+    def evaluate_condition_with_datasets(condition_string: str, datasets_filter: List[str]) -> Any:
+        """
+        Evaluates a condition using the dataset filter, transforming URIs into valid variable names.
+
+        :param condition_string: A string representing the logical condition to evaluate.
+            Example: "(dataset_custom_1 & dataset_custom_2) | dataset_custom_3".
+        :type condition_string: str
+        :param datasets_filter: A list of dataset URIs to be evaluated in the condition.
+        :type datasets_filter: List[str]
+
+        :returns: The result of the logical condition evaluation with URIs replaced by valid variable names.
+        :rtype: Any
+        """
+        dataset_map = {}
+        for uri in datasets_filter:
+            valid_variable_name = utils.make_valid_variable_name(uri)
+            condition_string = condition_string.replace(uri, valid_variable_name)
+            dataset_map[valid_variable_name] = Dataset(uri)
+        evaluated_condition = eval(condition_string, {}, dataset_map)
```

Comment:
The same concern I shared in another part of the code applies here, in regards to using `eval`.